no-redeclare broken for class static props in 6.1.1

Suddenly after an npm install I got slammed with this, I'm assuming from some semver-compatible update to eslint or something related.  Woohoo.
**this issue is not reproducible (i.e. everything works) with `babel-eslint@6.0.5`**

**What version of ESLint are you using?**
2.13.1
**What parser (default, Babel-ESLint, etc.) are you using?**
6.1.1
**Please show your full configuration:**
babel config
```
{
          "cacheDirectory": true,
          "presets": [
            "es2015",
            "stage-1",
            "react"
          ],
          "plugins": [
            "transform-flow-strip-types",
            "transform-decorators-legacy"
          ]
        }
```
`.eslintrc`
```
{
  "root": true,
  "parser": "babel-eslint",
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended"
  ],
  "rules": {
    "comma-dangle": 0,
    "strict": 0,
    "no-unused-vars": [1, {"args": "none", "varsIgnorePattern": "React"}],
    "no-console": 1,
    "react/display-name": 0,
    "react/no-unknown-property": 0,
    "react/no-did-mount-set-state": 0,
    "react/prop-types": 0,
    "comma-spacing": 2,
    "eol-last": 2,
    "keyword-spacing": 2,
    "linebreak-style": 2,
    "no-extra-semi": 2,
    "no-trailing-spaces": 2,
    "no-unexpected-multiline": 2,
    "no-unreachable": 2,
    "semi": ["error", "never"],
    "jsx-quotes": 2,
    "react/jsx-boolean-value": [2, "never"],
    "react/jsx-equals-spacing": [2, "never"],
    "react/jsx-closing-bracket-location": 2,
    "react/jsx-curly-spacing": [2, "never"],
    "react/jsx-indent-props": 2,
    "react/jsx-key": 2,
    "react/jsx-space-before-closing": 2
  },
  "env": {
    "meteor": true,
    "browser": true,
    "node": true,
    "es6": true,
    "commonjs": true
  },
  "globals": {
    "Roles": false
  },
  "plugins": [
    "react"
  ]
}
```
**What did you do? Please include the actual source code causing the issue.**
```
import React, {Children, Component, PropTypes, isValidElement, cloneElement} from 'react'
...

class Group extends Component {
  static propTypes = {
    ...
  };
  ...
}

export default class TypicalForm extends Component {
  static propTypes = {
    ...
  };
  ...
}
```
**What did you expect to happen?**
Code should lint just fine, as it was earlier today
**What actually happened? Please include the actual, raw output from ESLint.**
```
ERROR in /Users/andy/jcore-portal/jcore-react-components/common/TypicalForm.jsx

/Users/andy/jcore-portal/jcore-react-components/common/TypicalForm.jsx
  69:10  error  'propTypes' is already defined  no-redeclare

✖ 1 problem (1 error, 0 warnings)
```

--- a/index.js
+++ b/index.js
@@ -229,13 +229,6 @@ function monkeypatch() {
   // visit decorators that are in: ClassDeclaration / ClassExpression
   var visitClass = referencer.prototype.visitClass;
   referencer.prototype.visitClass = function(node) {
-    var classBody = node.body.body;
-    for (var a = 0; a < classBody.length; a++) {
-      if (classBody[a].type === "ClassProperty") {
-        createScopeVariable.call(this, classBody[a], classBody[a].key);
-      }
-    }
-
     visitDecorators.call(this, node);
     var typeParamScope;
     if (node.typeParameters) {
@@ -266,6 +259,14 @@ function monkeypatch() {
     }
     visitDecorators.call(this, node);
     visitProperty.call(this, node);
+  };
+
+  // visit ClassProperty as a Property.
+  referencer.prototype.ClassProperty = function(node) {
+    if (node.typeAnnotation) {
+      visitTypeAnnotation.call(this, node.typeAnnotation);
+    }
+    this.visitProperty(node);
   };
 
   // visit flow type in FunctionDeclaration, FunctionExpression, ArrowFunctionExpression

--- a/test/non-regression.js
+++ b/test/non-regression.js
@@ -1525,6 +1525,54 @@ describe("verify", function () {
     );
   });
 
+  describe("Class Property Declarations", function() {
+    it("no-redeclare false positive 1", function() {
+      verifyAndAssertMessages(
+        [
+          "class Group {",
+          "  static propTypes = {};",
+          "}",
+          "class TypicalForm {",
+          "  static propTypes = {};",
+          "}"
+        ].join("\n"),
+        { "no-redeclare": 1 },
+        []
+      );
+    });
+
+    it("no-redeclare false positive 2", function() {
+      verifyAndAssertMessages(
+        [
+          "function validate() {}",
+          "class MyComponent {",
+          "  static validate = validate;",
+          "}"
+        ].join("\n"),
+        { "no-redeclare": 1 },
+        []
+      );
+    });
+
+    it("check references", function() {
+      verifyAndAssertMessages(
+        [
+          "var a;",
+          "class A {",
+          "  prop1;",
+          "  prop2 = a;",
+          "  prop3 = b;",
+          "}",
+          "new A"
+        ].join("\n"),
+        { "no-undef": 1, "no-unused-vars": 1, "no-redeclare": 1 },
+        [
+          "5:11 'b' is not defined. no-undef"
+        ]
+      );
+    });
+  });
+
   // it("regex with es6 unicodeCodePointEscapes", function () {
   //   verifyAndAssertMessages(
   //     "string.replace(/[\u{0000A0}-\u{10FFFF}<>\&]/gmiu, (char) => `&#x${char.codePointAt(0).toString(16)};`);",

